src/pykeen/nn/init.py

```diff
@@ -120,6 +121,48 @@ def init_quaternions(
     return x.view(num_elements, 4 * dim)
 
 
+class PretrainedInitializer:
+    """
+    Initialize tensor with pretrained weights.
+
+    Example usage:
+
+    .. code-block::
+
+        import torch
+        from pykeen.pipeline import pipeline
+        from pykeen.nn.init import create_init_from_pretrained
+
+        # this is usually loaded from somewhere else
+        # the shape must match, as well as the entity-to-id mapping
+        pretrained_embedding_tensor = torch.rand(14, 128)
+
+        result = pipeline(
+            dataset="nations",
+            model="transe",
+            model_kwargs=dict(
+                embedding_dim=pretrained_embedding_tensor.shape[-1],
+                entity_initializer=PretrainedInitializer(tensor=pretrained_embedding_tensor),
+            ),
+        )
+    """
+
+    def __init__(self, tensor: torch.FloatTensor) -> None:
+        """
+        Initialize the initializer.
+
+        :param tensor:
+            the tensor of pretrained embeddings.
+        """
+        self.tensor = tensor
+
+    def __call__(self, x: torch.Tensor) -> torch.Tensor:
+        """Initialize the tensor with the given tensor."""
+        if x.shape != self.tensor.shape:
+            raise ValueError(f"shape does not match: expected {self.tensor.shape} but got {x.shape}")
```

Comment:
not for this PR, but we could consider exposing `self.tensor.shape`, and maybe allow providing `embedding_dim=None` if the initializer provides a shape.

Reply:
could you please elaborate?

Reply:
If we explicitly want to use an initializer from pre-trained embeddings, then we need to set `embedding_dim` manually to the matching dimension, e.g., we want to use label-based init from `bert-base-cased`, then we will get 768-dimensional vectors. Right now, we would either need to create the initializer beforehand and inspect the stored `tensor`'s shape, or would need to lookup the dimension of the `bert-base-cased` model to determine how to set the `embedding_dim`.

The suggestion above was to allow setting `embedding_dim=None` in a model, which will then use the initializer's shape. If the initializer does not have a fixed shape, an error would be raised.

Reply:
Oh that's a bit round-about. It might cause more pain than it's worth to implement that